appserver/tests/tck/microprofile/rest-client/src/main/java/org/glassfish/microprofile/rest/client/tck/client/BeansXmlTransformer.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2023 Contributors to Eclipse Foundation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.microprofile.rest.client.tck.client;
+
+import java.net.URL;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import org.jboss.arquillian.container.spi.event.container.BeforeDeploy;
+import org.jboss.arquillian.container.test.spi.client.deployment.ApplicationArchiveProcessor;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ArchivePath;
+import org.jboss.shrinkwrap.api.Node;
+import org.jboss.shrinkwrap.api.asset.ArchiveAsset;
+import org.jboss.shrinkwrap.api.asset.UrlAsset;
+
+import static java.lang.String.format;
+import static java.lang.System.Logger.Level.INFO;
+
+/**
+ * This extension replaces beans.xml files with ones declaring the 'all' bean discovery type.
+ * This is because version 3.0.1 of the TCK still deploys an empty beans.xml due to a faulty
+ * assumption that CDI < 4 is still defaulting to the 'all' type.
+ */
+public class BeansXmlTransformer implements ApplicationArchiveProcessor {
+
+    private static final System.Logger LOG = System.getLogger(BeansXmlTransformer.class.getName());
+
+    private static final String LIB_DIR_PATH = format("WEB-INF%slib", ArchivePath.SEPARATOR);
+
+    private static final String[] BEANS_XML_PATHS = {
+        format("/META-INF%sbeans.xml", ArchivePath.SEPARATOR),
```

Comment:
Unclear, so path can start with `/` but then its elements have to be separated with `AP.SEPARATOR`?

Reply:
@pzygielo, cite from the `org.jboss.shrinkwrap.api.ArchivePath` JavaDoc:

> Represents a target context within an [Archive](https://javadoc.io/static/org.jboss.shrinkwrap/shrinkwrap-api/2.0.0-beta-1/org/jboss/shrinkwrap/api/Archive.html) under which an [Node](https://javadoc.io/static/org.jboss.shrinkwrap/shrinkwrap-api/2.0.0-beta-1/org/jboss/shrinkwrap/api/Node.html) may be found. All [ArchivePath](https://javadoc.io/static/org.jboss.shrinkwrap/shrinkwrap-api/2.0.0-beta-1/org/jboss/shrinkwrap/api/ArchivePath.html) contexts are absolute (ie. prepended with the '/' character).

Thanks for comment, I fixed this in another place :)

Reply:
Looking into Shrinkwrap code, Archive.SEPARATOR is always '/': https://github.com/shrinkwrap/shrinkwrap/blob/main/api/src/main/java/org/jboss/shrinkwrap/api/ArchivePath.java#L31.

So it doesn't matter if we use '/' directly, or via the constant. But I wouldn't mix it. I think the Javadoc writes '/' but means ArchivePath.SEPARATOR, as there's no reason to use '/' in the beginning and something else later in the path.

I suggest the following code:

```
BEANS_XML_PATHS = {
        "/META-INF/beans.xml".replace('/', ArchivePath.SEPARATOR),
```

Or just use plain '/' because I don't believe hat the separator will ever change.

Reply:
I vote for just a simple `/` character without replace - it is just a hardcoded constant, so we are replacing `/` by `/` here.

Reply:
Done.

Many thanks all for advice.